### PR TITLE
Add location to coordinate clues

### LINKED ISSUE #7135: ADD DESCRIPTION TO COORDINATES CLUE - MEDIUMS
**Is your feature request related to a problem? Please describe.**
I'm always frustrated when i have to open the map every time when doing a lot of clues you waste a lot of time [...]

**Describe the solution you'd like**
Add description to coordinates map instead of https://imgur.com/a/SQGcdV4
**Describe alternatives you've considered**
ADD THIS - the list is not mine made by kelvino(list is public for everyone) but is complete and very accurate whit very good descriptions for each cord map+i consider those descriptions usefull only for medium clues beacause everyone is speedrunning meds for money nowdays
05.20S, 04.28E / Feldip Hills
01.35S, 07.28E / North of Hazelmere's house
00.05S, 01.13E / West of Tree Gnome Village
01.18S, 14.15E / North East Karamja
03.07S, 03.41W / South West of Castle Wars
00.31S, 17.43E / North of Mudskipper point
07.05N, 30.56E / Morytania Swamp
14.20N, 30.45W / South of Chasm of Fire
09.48N, 17.39E / Ice Mountain
12.28N, 34.37E / Fenkenstrain's Castle garden
00.30N, 24.16E / Lumbridge Swamp
00.20S, 23.15E / South of Water Alter in Lumbridge Swamp
00.18S, 09.28E / South of Brimhaven dungeon
01.26N, 08.01E / Moss Giant Island, West of Brimhaven
04.00S, 12.46E / Tai Bwo Wannai mine
03.35S, 13.35E / East of Tai Bwo Wannai
06.41N, 27.15E / West of Digsite Library
11.33N, 02.24W / South of the Falconer, near Eagle's Peak
09.35N, 01.50W / North of Terrorbird pen, Gnome Stronghold
02.48N, 22.30E / West of H.A.M hideout
05.43N, 23.05E / Lumbridge cow pen
00.13S, 13.58E / West of Banana Plantation on Karamja
04.13N, 12.45E / Crandor Island
11.41N, 14.58E / Burthorpe pub garden
10.45N, 04.31E / By Coal Trucks, North West of Ranging Guild
06.58N, 21.16E / North East of Draynor Manor
08.33N, 01.39W / Terrorbird pen, Gnome Strongold
11.05N, 00.45W / Grand Tree Swamp
11.03N, 31.20E / South of Morytania Slayer Tower, Outside gate area
11.18N, 30.54E / South of Morytania Slayer Tower
14.54N, 9.13E / North West of the Golden Apple Tree
06.31N, 01.46W / West of Outpost
0.243S, 33.26E / South of Uzer
22.30N, 03.01E / North of Miscellania teleport
02.50N, 06.20E / South of South Ardougne bank
15.22N, 07.31E / By the cabbage and onions
09.33N, 02.15E / Before the Waterfall entrance, South Barbarian Outpost
07.33N, 15.00E / Lady of the Lake in Taverly
08.11S, 04.48E / East of red chinchompas

### PR DESCRIPTION
**Pull request by WoneTooPhree**
Added location in the infobox for all coordinate clues.
Followed the same pattern as Cryptic, Anagram,... code.

Opening the world map take a bit of time (also freeze the client for less than a second) and searching all corner of world map to see the clue scroll icon take even more time.
Before/After
coordinateclue

I try to keep the text location as small as possible.
Getting to the location should make the hint arrow appears on screen (unless setting is off ? ).

In previous code it says that it is possible for a clue to say "degree" instead of "degrees" (Wiki use "degrees" for every clue).
Found a video (from 2016) of 1 clue, in game is "1 degree 54 minutes south 08 degrees 54 minutes west".
I can't test it.
I don't know if Jagex changed it to "01 degrees..."

But in case some clues don't have the same text then it will convert the coordinate to world location and tell the user the open the world map like it is right now.
Also usefull if in the next clue scroll expension they add more coordinate clue.

I'm a beginner, I did mostly copy/paste so if you could help me improve/clean it. 
Or if you have a better text location for any clue, tell me.

Thank you.

Closes #7135

### CODE REVIEW

**Comment by deathbeam** [comment]
Definitely do not remove parsing of the location. so, keep it and adjust your code around parsed locations

**Comment by WoneTooPhree** [comment]
Can you tell me if it's the correct way of doing it ? 

`solution = CLUES.get(location);`

Put coordinatesToWorldPoint() back to his original file.
Also looks like i forgot a new line at the end of file...

File: runelite-client/src/main/java/net/runelite/client/plugins/cluescrolls/clues/CoordinateClue.java [hunk 1]
```diff
@@ -43,18 +45,182 @@
...
+		put(new WorldPoint(1442, 3878, 0), "Sulphur Mine.");
+		put(new WorldPoint(3380, 3929, 0), "Wilderness. Near Volcano.");
+		put(new WorldPoint(3188, 3939, 0), "Resource Area.");
+		put(new WorldPoint(3304, 3941, 0), "East of Rogues' Castle.");
+		put(new WorldPoint(2994, 3961, 0), "Wilderness. Inside Agility Training Area.");
+	}};
+
 	private String text;
 	private WorldPoint location;
+	private static String solution;
```
Comment (raiyni) [inline runelite-client/src/main/java/net/runelite/client/plugins/cluescrolls/clues/CoordinateClue.java]: You don't need to predefine this nor make it static

File: runelite-client/src/main/java/net/runelite/client/plugins/cluescrolls/clues/CoordinateClue.java [hunk 2]
```diff
@@ -43,18 +45,182 @@
 @AllArgsConstructor
 public class CoordinateClue extends ClueScroll implements TextClueScroll, LocationClueScroll
 {
+	public static final Map<WorldPoint, String> CLUES = new HashMap<WorldPoint, String>()
```
Comment (raiyni) [inline runelite-client/src/main/java/net/runelite/client/plugins/cluescrolls/clues/CoordinateClue.java]: Use `ImmutableMap::builder`

Comment (raiyni) [reply runelite-client/src/main/java/net/runelite/client/plugins/cluescrolls/clues/CoordinateClue.java]: also make it private

File: runelite-client/src/main/java/net/runelite/client/plugins/cluescrolls/ClueScrollPlugin.java [hunk 3]
```diff
@@ -668,4 +668,4 @@ private void updateClue(final ClueScroll clue)
 		checkClueNPCs(clue, client.getCachedNPCs());
 		this.clue = clue;
 	}
-}
+}
```
Comment (raiyni) [inline runelite-client/src/main/java/net/runelite/client/plugins/cluescrolls/ClueScrollPlugin.java]: undo this change

File: runelite-client/src/main/java/net/runelite/client/plugins/cluescrolls/clues/CoordinateClue.java (hunk 2 above)
Comment (WoneTooPhree) [reply runelite-client/src/main/java/net/runelite/client/plugins/cluescrolls/clues/CoordinateClue.java]: Can you tell me if its the right way of doing ImmutableMap.
ty

Comment (Nightfirecat) [reply runelite-client/src/main/java/net/runelite/client/plugins/cluescrolls/clues/CoordinateClue.java]: Looks right to me. :+1:

File: runelite-client/src/main/java/net/runelite/client/plugins/cluescrolls/clues/CoordinateClue.java [hunk 4]
```diff
@@ -52,9 +201,23 @@ public void makeOverlayHint(PanelComponent panelComponent, ClueScrollPlugin plug
...
-			.build());
+		String solution = CLUES.get(location);
+
+		if (solution != null)
+		{
+			panelComponent.getChildren().add(LineComponent.builder()
+				.left(solution)
+				.build());
+
+			panelComponent.getChildren().add(LineComponent.builder().left("").build());
```
Comment (Adam-) [inline runelite-client/src/main/java/net/runelite/client/plugins/cluescrolls/clues/CoordinateClue.java]: Is this the best way to get an empty line?

Comment (Nightfirecat) [reply runelite-client/src/main/java/net/runelite/client/plugins/cluescrolls/clues/CoordinateClue.java]: Apparently it is. We have handling for `<br>` in tooltips, but not in LineComponents or TitleComponents. This is the same as how the clue scroll plugin adds "Spade required" text: https://github.com/runelite/runelite/blob/master/runelite-client/src/main/java/net/runelite/client/plugins/cluescrolls/clues/MapClue.java#L165-L166

File: runelite-client/src/main/java/net/runelite/client/plugins/cluescrolls/clues/CoordinateClue.java [hunk 5]
```diff
@@ -52,9 +201,23 @@ public void makeOverlayHint(PanelComponent panelComponent, ClueScrollPlugin plug
...
+			panelComponent.getChildren().add(TitleComponent.builder().text("Marked on world map.").build());
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/cluescrolls/clues/CoordinateClue.java]: You should use same message in both cases and in case there is solution just prepend those 2 lines above to it.

Comment (WoneTooPhree) [reply runelite-client/src/main/java/net/runelite/client/plugins/cluescrolls/clues/CoordinateClue.java]: I'm not sure to understand.
But if we don't know the coordinate, show the default message.
And if we know the coordinate show the default message + the solution ? 
CoordinateClue

Comment (deathbeam) [reply runelite-client/src/main/java/net/runelite/client/plugins/cluescrolls/clues/CoordinateClue.java]: Yeah exactly. Maybe alter default message to be shorter

Comment (WoneTooPhree) [reply runelite-client/src/main/java/net/runelite/client/plugins/cluescrolls/clues/CoordinateClue.java]: I think the actual message is fine, I make sure to not write to much in the solution.
I did a master clue scroll and got the jogre dig spot but the solution was so long that it's impossible to read all of it on fixed mode so i try to make the solution smaller than that one.

But still I think the number of plugins on and at the same spot as the clue helper can make it go out of screen.

Anyway here with smaller default solution (3 lines instead of 5).
CoordinateClue

Choose the one you prefer and I'll update the PR.

Thanks

Comment (deathbeam) [reply runelite-client/src/main/java/net/runelite/client/plugins/cluescrolls/clues/CoordinateClue.java]: I would put the detailed description at start, otherwise it looks fine.

Comment (deathbeam) [reply runelite-client/src/main/java/net/runelite/client/plugins/cluescrolls/clues/CoordinateClue.java]: Any update on this @WoneTooPhree ?

Comment (WoneTooPhree) [reply runelite-client/src/main/java/net/runelite/client/plugins/cluescrolls/clues/CoordinateClue.java]: Sorry here are the changes.

File: runelite-client/src/main/java/net/runelite/client/plugins/cluescrolls/clues/CoordinateClue.java [hunk 6]
```diff
@@ -52,8 +201,18 @@ public void makeOverlayHint(PanelComponent panelComponent, ClueScrollPlugin plug
...
+		String solution = CLUES.get(location);
+
+		if (solution != null)
+		{
+			panelComponent.getChildren().add(LineComponent.builder()
+				.left(solution)
+				.build());
+		}
+
+		panelComponent.getChildren().add(LineComponent.builder().build());
```
Comment (Adam-) [inline runelite-client/src/main/java/net/runelite/client/plugins/cluescrolls/clues/CoordinateClue.java]: This probably shouldn't be added unless the solution was added..?

**Comment by WoneTooPhree** [comment]
Updated
